gc/base/Dispatcher.cpp

```diff
@@ -108,3 +110,9 @@ void
 MM_Dispatcher::shutDownThreads() 
 {
 }
+
+uintptr_t
+MM_Dispatcher::recomputeActiveThreadCount(MM_EnvironmentBase *env, MM_Task *task, uintptr_t newThreadCount){
+	Assert_MM_unreachable();
```

Comment:
This should not assert.  IF a GC was not using the parallel Dispatcher then it has to fall to the Dispatcher version.  I am fine with this just returning 1 with no assert.

Reply:
It would be more correct to just return `_activeThreadCount` rather than 1. But that's not possible without another rework.... As Alek mentioned no collector instantiates dispatcher it self, so the little implementation that Dispatcher has can be moved to PD.

Most of the common dispatcher code seems to be in PD anyways, such as
```C++
uintptr_t _threadCountMaximum; /**< maximum threadcount - this is the size of the thread tables etc */
uintptr_t _threadCount; /**< number of threads currently forked */
uintptr_t _activeThreadCount; /**< number of threads actively running a task */
```
I don't thin Dispatcher should even have any idea of `recomputeActiveThreadCount` when it doesn't have the `_activeThreadCount` info. However, it had to be added here since Dispatcher run method is used by PD.

Reply:
Agreed.  The change is ok as-is.  The dispatcher / parallel dispatcher should be re-worked in a separate PR.  Thanks.

Reply:
I think a larger rework is required in general to clean all this up, an issue beyond what this PR addresses. A separate issue should be opened to do a larger refactor/cleanup in addition to what Alek pointed out below

Reply:
I'll make the change to remove the Assert and return 1... it's more consistent with the other Dispatcher methods 

 `dispatcher.hpp`
```C++
	MMINLINE virtual uintptr_t threadCount() { return 1; }
	MMINLINE virtual uintptr_t threadCountMaximum() { return 1; }
	MMINLINE virtual uintptr_t activeThreadCount() { return 1; }
```